specs.js

```diff
@@ -112,75 +135,571 @@ describe('when using trimperson api', () => {
+      })
+
+      When((done) => {
+        trim.getDocument(documentId, function (err, data) {
           dataReturn = data
+          errReturn = err
           done()
         })
       })
-      Then('it should be able to succesfully use the function', () => {
+
+      Then('It should return an error', (done) => {
+        expect(errReturn).to.exist
+        expect(dataReturn).not.to.exist
+        getDocumentMock.done()
+        done()
+      })
+    })
+  })
+
+  /*
+   CREATE CONTAINER
+   */
+  describe('To create a container', () => {
+    Then('`createContainer` should be a function', (done) => {
+      expect(trim.createContainer).to.be.a('function')
+      done()
+    })
+    Then('`createPublicContainer` should be a function', (done) => {
+      expect(trim.createPublicContainer).to.be.a('function')
+      done()
+    })
+    Then('`createPrivateContainer` should be a function', (done) => {
+      expect(trim.createPrivateContainer).to.be.a('function')
+      done()
+    })
+
+    describe('If all inputs are valid', () => {
+      var createContainerMock, dataReturn, errReturn, data
+
+      Given(() => data = {
+        folderName: VALID_FOLDER_NAME,
+        privacyLevel: trim.PRIVACY_LEVELS.PUBLIC
+      })
+      Given(() => {
+        createContainerMock = nock(mockURL)
+            .filteringRequestBody(R.always('*'))
+            .post('/CreateContainer', '*')
+            .query({securityToken: mockToken})
+            .reply(201, {
+              RecordNo: VALID_FOLDER_NAME
+            })
+      })
+      When(done => {
+        trim.createContainer(data, (err, body) => {
+          errReturn = err
+          dataReturn = body
+          done()
+        })
+      })
+      Then('The container is created', done => {
+        expect(errReturn).not.to.exist
+        expect(dataReturn).to.be.a('object')
+        expect(dataReturn.RecordNo).to.equal(VALID_FOLDER_NAME)
+        createContainerMock.done()
+        done()
+      })
+    })
+
+    describe('If all no privacy is specified', () => {
+      var createContainerMock, dataReturn, errReturn, data
+
+      Given(() => data = {
+        folderName: VALID_FOLDER_NAME
+      })
+      Given(() => {
+        createContainerMock = nock(mockURL)
+            .filteringRequestBody(R.always('*'))
+            .post('/CreateContainer', '*')
+            .query({securityToken: mockToken})
+            .reply(201, {
+              RecordNo: VALID_FOLDER_NAME
+            })
+      })
+      When(done => {
+        trim.createContainer(data, (err, body) => {
+          errReturn = err
+          dataReturn = body
+          done()
+        })
+      })
+      Then('The container is created', done => {
+        expect(errReturn).not.to.exist
+        expect(dataReturn).to.be.a('object')
+        expect(dataReturn.RecordNo).to.equal(VALID_FOLDER_NAME)
+        createContainerMock.done()
+        done()
+      })
+    })
+
+    describe('No folder name is provided', () => {
```

Comment:
Yeah fair point. I noticed that TRIM doesn't like colons for document names and that might apply to container names as well. Maybe `INVALID_FOLDER_NAME` should be a name with a colon in it instead of an empty one.